src/ui/ResultsPerPage/ResultsPerPage.ts

```diff
@@ -216,6 +220,11 @@ export class ResultsPerPage extends Component {
     $$(this.span).addClass('coveo-results-per-page-no-results');
     $$(this.list).empty();
   }
+
+  private hideIfResultListHasInfiniteScrollEnabled() {
```

Comment:
Should not only hide, but also `disable()` the component, otherwise it is needlessly performing operations in the background/handling events etc.

We need a verbose info message that explains why it's getting hidden/disabled

Also need to add a warning and handle when we detect that there is only one result list in infinite scroll mode (this is by far what the vast majority of client will do), and that we are detecting a useless pager/useless results per page component. The warning message needs to be extremely specific in how to resolve the problem (remove it from the page I'm guessing).

When you disable the component, it also means that the `bind.onRootElement` function will be unable to perform it's function once disabled. 

You'll need to use a simple event listener without the `bind` wrapper.

Same thing for the pager.